functionary/core/utils/tasking.py

```diff
@@ -132,3 +136,14 @@ def _update_task_status(task: Task, status: int) -> None:
 
     if task.scheduled_task is not None and status == "ERROR":
         task.scheduled_task.error()
+
+
+def _handle_workflow_run(workflow_run: WorkflowRun, task: Task) -> None:
+    """Start the next task for a WorkflowRun or update its status as appropriate"""
+    if task.status == Task.COMPLETE:
+        next_task = workflow_run.execute_next_step()
```

Comment:
This function could be simplified to just being a `match` case statement, which may improve readability in the future if we add more task statuses.  

The `next_task` conditional could use the walrus operator so you don't need the separate conditional block if `next_task` is `None`